ibis/expr/tests/test_value_exprs.py

```diff
@@ -100,6 +101,64 @@ def test_literal_with_different_type(value, expected_type):
 
 
 @pytest.mark.parametrize(
+    ['value', 'expected_type', 'expected_class'],
+    [
+        (list('abc'), 'array<string>', ir.ArrayScalar),
+        ([1, 2, 3], 'array<int8>', ir.ArrayScalar),
+        ({'a': 1, 'b': 2, 'c': 3}, 'map<string, int8>', ir.MapScalar),
+        ({1: 2, 3: 4, 5: 6}, 'map<int8, int8>', ir.MapScalar),
```

Comment:
Aside: I wonder how much we benefit / are hurt overall from aggressively choosing the smallest type integer that will fit the literal values vs. using int32 (unless some values are larger than INT32_MAX)

Reply:
Interesting, I tried to compute `(2 ** 20) ** 200000` and got an overflow error. I think we should not try to infer the smallest integer type containing the result of an integer operation since it could take an unreasonably long amount of time to raise `OverflowError`.

Reply:
Yeah, the question is what is the benefit or cost (if any) of `literal([1, 2, 3])` returning an `array<int8>` vs. `array<int32>`